payments/providers/base.py

```diff
@@ -59,7 +80,7 @@ def ui_redirect_success(cls, return_url: str, order: Order = None) -> HttpRespon
         This should be used after a successful payment instead of the
         standard Django redirect.
         """
-        return cls._redirect_to_ui(return_url, 'success', order)
+        return cls._redirect_to_ui(return_url, 'success', order) if return_url else HttpResponse(status=204)
```

Comment:
The Respa 204 response could contain at least some text like "Payment success but error redirecting back to UI" or something. It won't be localized nor perhaps understandable to most of the end-users, but still better than a blank screen IMO.

Reply:
Yea you're right, although these cases where return url is missing should rarely happen it doesn't hurt if there's some message attached.